app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java

```diff
@@ -135,25 +127,23 @@ public void unzipFile() {
         SharedPreferences.Editor editor = sharedPref.edit();
         try {
             for (DownloadExpansionFile.XAPKFile xf : xAPKS) {
-                expansionFilePath = getExpansionFilePath(xf.mIsMain, xf.mFileVersion);
-                expansionFile = new File(expansionFilePath);
-                expansionZipFile = new ZipFile(expansionFile);
-                _zip = new Zip(expansionZipFile, this);
-                unzipFilePath = getUnzippedExpansionFilePath();
-                packageNameDir = new File(unzipFilePath);
-                if (xf.mIsMain) {
-                    if (packageNameDir.exists()) {
-                        DownloadExpansionFile.deleteDir(packageNameDir);
+                if (xf.mIsMain && xf.mFileVersion != storedMainFileVersion || !xf.mIsMain && xf.mFileVersion != storedPatchFileVersion) {
+                    expansionFilePath = getExpansionFilePath(xf.mIsMain, xf.mFileVersion);
+                    expansionFile = new File(expansionFilePath);
+                    expansionZipFile = new ZipFile(expansionFile);
+                    _zip = new Zip(expansionZipFile, this);
+                    unzipFilePath = getUnzippedExpansionFilePath();
+                    packageNameDir = new File(unzipFilePath);
+                    if (xf.mIsMain && !packageNameDir.exists()) {
+                        packageNameDir.mkdir();
+                    }
+                    _zip.unzip(unzipFilePath, totalZipSize);
+                    _zip.close();
+                    if (xf.mIsMain) {
```

Comment:
What if the extraction process fails? Do we need to save file version

Reply:
Updated. Stored file version before exception occurs.